libs/mock-builder/tests/pallet.rs

```diff
@@ -190,4 +210,48 @@ mod test {
 		// The storage is dropped at this time. Mocks no longer found from here.
 		mock_not_configured();
 	}
+
+	#[test]
+	fn generic_input() {
+		new_test_ext().execute_with(|| {
+			MockAB::mock_generic_input(|p1: i8, p2: u8| {
+				assert_eq!(p1, 1);
+				assert_eq!(p2, 2);
+				8
+			});
+			MockAB::mock_generic_input(|p1: i16, p2: u16| {
+				assert_eq!(p1, 3);
+				assert_eq!(p2, 4);
+				16
+			});
+
+			assert_eq!(MockAB::generic_input(1i8, 2u8), 8);
+			assert_eq!(MockAB::generic_input(3i16, 4u16), 16);
+		});
+	}
+
+	#[test]
+	#[should_panic]
```

Comment:
Same as above: I would prefer to narrow down the panic.

Reply:
This panic comes from a bad macro usage, and I can not check from the test that "bad usage". Only I can expect it to panic at some point in the macro.

But this panic is ok from a use-case perspective. Tells the user they have configured its mock badly.